sys/net/network_layer/ng_sixlowpan/ng_sixlowpan.c

```diff
@@ -78,13 +78,20 @@ void _receive(ng_pktsnip_t *pkt)
         /* packet is uncompressed: just mark and remove the dispatch */
         sixlowpan = ng_pktbuf_add(payload, payload->data, sizeof(uint8_t),
                                   NG_NETTYPE_SIXLOWPAN);
-        LL_DELETE(pkt, sixlowpan);
-        ng_pktbuf_release(sixlowpan);
+
+        if (sixlowpan == NULL) {
+            DEBUG("6lo: can not mark 6LoWPAN dispatch\n");
+            ng_pktbuf_release(pkt);
+            return;
+        }
+
+        pkt = ng_pktbuf_remove_snip(pkt, sixlowpan);
     }
 #ifdef MODULE_NG_SIXLOWPAN_FRAG
     else if (ng_sixlowpan_frag_is((ng_sixlowpan_frag_t *)dispatch)) {
         DEBUG("6lo: received 6LoWPAN fragment\n");
         ng_sixlowpan_frag_handle_pkt(pkt);
+        return;
```

Comment:
because otherwise the function continues with the fragment data, assuming it is a full packet (=> sending it to IPv6)